src/NuGet.Services.Validation.Orchestrator/ContentScan/ContentScanValidator.cs

```diff
@@ -0,0 +1,155 @@
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            var validatorStatus = await GetValidatorStatusAsync(request);
+
+            if (validatorStatus.State != ValidationStatus.NotStarted)
+            {
+                _logger.LogWarning(
+                    "Scan and Sign validation with validation Id {ValidationStepId} has already started.",
+                    request.ValidationStepId);
+
+                return GetValidationResponse(validatorStatus);
+            }
+
+            if (ShouldSkipScan(request))
+            {
+                return ValidationResponse.Succeeded;
+            }
+
+            await _contentScanEnqueuer.EnqueueContentScanAsync(request.ValidationStepId, request.InputUrl.OriginalString);
+
+            var result = await _validatorStateService.TryAddValidatorStatusAsync(request, validatorStatus, ValidationStatus.Incomplete);
+
+            var response = GetValidationResponse(result);
+            return response;
+        }
+
+        public async Task CleanUpAsync(IValidationRequest request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+            //TODO: need to check if cvs blobs should be clear here or by CVSJob(prefer)
+        }
+
+        public async Task<IValidationResponse> GetResponseAsync(IValidationRequest request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            var result = await GetValidatorStatusAsync(request);
+
+            var response = GetValidationResponse(result);
+            return response;
+        }
+
+        private async Task<ValidatorStatus> GetValidatorStatusAsync(IValidationRequest request)
+        {
+            var validatorStatus = await _validatorStateService.GetStatusAsync(request);
+            return validatorStatus;
+        }
+
+        private bool ShouldSkipScan(IValidationRequest request)
+        {
+            return false;
+        }
+
+        private IValidationResponse GetValidationResponse(ValidatorStatus status)
+        {
+            //assuming only terminal states will have packageValidationResult
+            if(status.State == ValidationStatus.Incomplete || 
+                status.State == ValidationStatus.NotStarted)
+            {
+                return new ValidationResponse(status.State);
+            }
+
+            var packageValidationresult = new PackageValidationResult();
+            switch (status.State)
+            {
+                case ValidationStatus.Succeeded:
+                    packageValidationresult.Type = "ContentScanSucceed";
+                    break;
+                case ValidationStatus.Failed:
+                    packageValidationresult.Type = "ContentScanFailed";
+                    packageValidationresult.Data = "Content Scan failed";
```

Comment:
Here's the section in the protocol that is relevant here: https://github.com/NuGet/Engineering/blob/master/Server.Specs/ValidationProtocol.md#content-violation

```suggestion
                    packageValidationresult.Type = "ContentViolation";
                    packageValidationresult.Data = "{}";
```